Scripts/Commands: Fix trinity_string usage and enhance commands

Rochet2 opened the pull request:
[//]: # (***************************)
[//]: # (** FILL IN THIS TEMPLATE **)
[//]: # (***************************)

**Changes proposed:**

~~Waiting for https://github.com/TrinityCore/TrinityCore/pull/18978 to merge first!~~

Tries to fix trinity_string related format differences and more:
- Change many %d to %u
- Change many %i to %u
- Change many %u and %llu to %s and change uint64 values to std::to_string(val).c_str()
- Change int64 to string before formatting and change all format characters for them to %s
- Change uint64 to string before formatting and change all format characters for them to %s
- Change many atoi to atoul when trying to get an uint32 value from user
- Change many atoi to atoul when trying to get an uint32 value from user
- Add .c_str() call to convert std::string to const char* before formatting
- Remove format characters that were not used in code (in DB there was %s but nothing in c++)
- Remove arguments that had no format characters (in c++ there was something formatted, but in DB there was no format character like %s)
- Change types in code to match the surrounding code better (for example return type was uint32 and returned variable was int)
- Remove unnecessary `(char*)` casts for `const char*` variables
- `.modify faction` now works without arguments as described in `.help`
- LANG_COMMAND_DELOBJREFERCREATURE (274) no longer says that a nonexistant creature references it, because owner can be more than a creature. Instead it says that a nonexistant owner references it. Printed guid should tell what type the owner is.
- Change __some__ usages of snprintf to use Trinity::StringFormat instead
- Fix typos (IE: Wypoint scipt -> Waypoint script)
- Convert __some__ `container.size()` to string before formatting
- Change __some__ std::strtoull to atoull

**Target branch(es):** 3.3.5/master

- [x] 3.3.5
- [x] master

**Issues addressed:** Closes #  (insert issue tracker number)
Closes #19127
Closes #20023

**Tests performed:** (Does it build, tested in-game, etc.)
Does build.

**Known issues and TODO list:** (add/remove lines as needed)

- [x] Fix the code (the IF condition here looks funny) and the message (the guids are in wrong order) and the types in here: https://github.com/TrinityCore/TrinityCore/pull/18978#discussion_r100573183 (usage of LANG_COMMAND_DELOBJREFERCREATURE in cs_gobject.cpp)
- [x] Fix the PR after https://github.com/TrinityCore/TrinityCore/pull/18978 is merged.

Rochet2 committed:
Scripts/Commands: Fix trinity_string usage

- Change many %d to %u
- Change many %u and %llu to %s and change uint64 values to std::to_string(val).c_str()
- Change many %i to %u
- Remove some format characters that were not used in code
- Removed some arguments that had no format characters
- Changed some types in code to match the surrounding code better (for example return value was uint32 and variable was int)
- Changed many atoi to atoul when trying to get an uint32 value from user
- Removed unnecessary (char*) casts
- Changed some usages of snprintf to use Trinity::StringFormat instead
- Change int64 to string before formatting and change all format characters for them to %s
- Add .c_str() call to convert std::string to const char* before formatting
- Use static_cast to cast some list.size() calls to uint32 before formatting

Aokromes commented on sql/updates/world/master/xxxx_xx_xx_x1_world.sql:
```diff
@@ -0,0 +1,97 @@
+UPDATE `trinity_string` SET `content_default`='Up to %i expansions are allowed now.' WHERE  `entry`=61;
```
merge this world file with previous file.

Rochet2 replied:
" I hope these commits are not merged together. This is because of the first being a cherry pick of 3.3.5 branch commit and the second is more work built on top of that."

Still want it though?

Aokromes replied:
then 2 different PRs is the way, we cannot partialy merge PRs.

Rochet2 replied:
The PRs depend on each other though so making separate PRs now would make the other have conflicts with master so not sure that is possible in a good way.

Guess I need to merge them then if there is no other way.

Aokromes replied:
you can create 2 separate PRs and fix conflicts when the 1st is merged.

Kittnz commented:
needs a bit of updating it seems

Rochet2 commented:
> needs a bit of updating it seems

Added more info to todo and top of PR description. Waiting on other PR to fix this one.

Shauren commented:
The other one is now merged

Rochet2 committed:
Fix modify faction behavior to help text

Help text says .mod faction should display current status when no arguments given.
However in that case it immediately returns false. Removed this check.

Also changed some atoi to atoul and strtoull to atoull.

Rochet2 committed:
Fix LANG_COMMAND_DELOBJREFERCREATURE

Change LANG_COMMAND_DELOBJREFERCREATURE (274)
Original: Game Object (GUID: %s) is referenced by nonexistent creature %s in GO list, can't be deleted.
New: Game Object (GUID: %s) has a nonexistent owner %s, can't be deleted.

Rochet2 commented:
Would be cool if we could ignore the type of variables when printing them.
Using something like
http://pastebin.com/aYrebMhy
or http://fmtlib.net/latest/api.html#format-api

Rochet2 committed:
Fix typo

Rochet2 commented:
Updated PR and descriptions.

Commit:
Update and rename xxxx_xx_xx_x1_world.sql to 2017_99_99_18863_world.sql

- added comment on line 1

Rochet2 committed:
Merge branch 'master' of github.com:TrinityCore/TrinityCore into fixcommands

Rochet2 commented:
Fixed the bad merge

jackpoz commented:
merge this to stop users from reporting the same crashes over and over ?

Aokromes commented:
3.3.5a version of this was merged months ago, this fixes at least few crashes......

jackpoz commented:
merge it and hope for the best

Rochet2 commented:
@Aokromes What PR are you referring to?
If it is one of mine, then https://github.com/TrinityCore/TrinityCore/pull/18978 and https://github.com/TrinityCore/TrinityCore/pull/18278 are the same PR but the first one is a cherry pick to from 3.3.5 to master branch.

Those PRs were attempting to fix gameobject targetting which did not work and when it worked it was working in an unreliable manner (possibly selecting unwanted gobs).

This PR we are in tries to solve crashes and issues with commands trying to use the new lowguids that are uint64. This is solved by conversion to string so we can insert the numbers easily to commands.

Just making sure if something was unclear.
I guess fixing the crashes would be nice :)
